### PR TITLE
Change JDL to return applications with entities.

### PR DESCRIPTION
**Pull request by mshima**
- Unify skipYoRcGeneration and skipEntityFilesGeneration into skipConfigFilesGeneration

Related to https://github.com/jhipster/generator-jhipster/issues/12178

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by MathieuAA** [comment]
"skipConfigFilesGeneration" why not simply skipFileGeneration?

**Comment by mshima** [comment]
@MathieuAA done.

File: jdl/exporters/jhipster-entity-exporter.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ module.exports = {
  * @param passedConfiguration the object having the keys:
  *        - entities: the entities to exporters,
  *        - forceNoFiltering: whether to filter out unchanged entities,
- *        - skipEntityFilesGeneration: whether to skip file write to disk,
```
Comment (MathieuAA) [inline jdl/exporters/jhipster-entity-exporter.js:41]: The function is deprecated, I'll be removing them in a PR soon. No need to update it.

File: test/jdl/jdl-importer.spec.js [hunk 2]
```diff
@@ -567,6 +568,18 @@ relationship OneToOne {
                 expect(fse.statSync('.jhipster').isDirectory()).to.be.true;
                 expect(fse.statSync(path.join('.jhipster', 'BankAccount.json')).isFile()).to.be.true;
             });
+            it('returns corresponding exportedApplicationsWithEntities', () => {
```
Comment (MathieuAA) [inline test/jdl/jdl-importer.spec.js:571]:
```suggestion
            it('should return the corresponding exportedApplicationsWithEntities', () => {
```

File: test/jdl/jdl-importer.spec.js [hunk 3]
```diff
@@ -607,6 +620,18 @@ relationship OneToOne {
...
```
Comment (MathieuAA) [inline test/jdl/jdl-importer.spec.js:623]:
```suggestion
            it('should return the corresponding exportedApplicationsWithEntities', () => {
```

File: test/jdl/jdl-importer.spec.js [hunk 4]
```diff
@@ -633,6 +658,18 @@ relationship OneToOne {
                 expect(content['generator-jhipster'].entitySuffix).to.equal('Entity');
                 expect(content['generator-jhipster'].dtoSuffix).to.equal('DTO');
             });
+            it('returns corresponding exportedApplicationsWithEntities', () => {
```
Comment (MathieuAA) [inline test/jdl/jdl-importer.spec.js:661]:
```suggestion
            it('should return the corresponding exportedApplicationsWithEntities', () => {
```

File: test/jdl/jdl-importer.spec.js [hunk 5]
```diff
@@ -1034,6 +1071,18 @@ relationship OneToOne {
                     }
                 });
             });
+            it('returns corresponding exportedApplicationsWithEntities', () => {
```
Comment (MathieuAA) [inline test/jdl/jdl-importer.spec.js:1074]:
```suggestion
            it('should return the corresponding exportedApplicationsWithEntities', () => {
```

File: test/jdl/jdl-importer.spec.js [hunk 6]
```diff
@@ -1780,7 +1829,7 @@ paginate * with infinite-scroll
                 expect(entityF).to.be.false;
             });
         });
-        context('when passing skipYoRcGeneration and skipEntityFilesGeneration options', () => {
+        context('when passing skipFileGeneration options', () => {
```
Comment (MathieuAA) [inline test/jdl/jdl-importer.spec.js:1832]:
```suggestion
        context('when passing the skipFileGeneration option', () => {
```

File: test/jdl/jdl-importer.spec.js [hunk 7]
```diff
@@ -504,6 +504,7 @@ describe('JDLImporter', () => {
                     ],
                     exportedApplications: [],
                     exportedDeployments: [],
+                    exportedApplicationsWithEntities: {},
```
Comment (MathieuAA) [inline test/jdl/jdl-importer.spec.js:507]: Why do we need it?

File: test/jdl/exporters/jhipster-application-exporter.spec.js [hunk 8]
```diff
@@ -319,6 +319,45 @@ describe('JHipsterApplicationExporter', () => {
...
+                        },
+                        { skipFileGeneration: true }
+                    );
+                });
+
+                it('should return the exported applications', () => {
+                    expect(returned).to.have.lengthOf(2);
+                });
+                it('should not save config files', () => {
+                    const baseNames = returned.map(application => application['generator-jhipster'].baseName);
```
Comment (MathieuAA) [inline test/jdl/exporters/jhipster-application-exporter.spec.js:353]: Put it in the `before block`

File: jdl/exporters/jhipster-entity-exporter.js [hunk 9]
```diff
@@ -38,7 +38,7 @@ module.exports = {
...
+ *        - skipFileGeneration: whether to skip file write to disk,
```
Comment (mshima) [inline jdl/exporters/jhipster-entity-exporter.js:41]:
```suggestion
 *        - skipEntityFilesGeneration: whether to skip file write to disk,
```

File: test/jdl/jdl-importer.spec.js (hunk 7 above)
Comment (mshima) [reply test/jdl/jdl-importer.spec.js:507]: I am moving save/formatting to the generator.
https://github.com/jhipster/generator-jhipster/issues/12178

**Comment by mshima** [comment]
> If you wanna do it, by all means! I can do it right after this PR :)

You should do it, you are the jdl export :)

**Comment by mshima** [comment]
Thanks @MathieuAA

**Pull request merged**